api/src/main/java/org/pmiops/workbench/db/dao/UserRecentWorkspaceDao.java

```diff
@@ -0,0 +1,12 @@
+package org.pmiops.workbench.db.dao;
+
+import java.util.List;
+import java.util.Optional;
+import org.pmiops.workbench.db.model.UserRecentWorkspace;
+import org.springframework.data.repository.CrudRepository;
+
+public interface UserRecentWorkspaceDao extends CrudRepository<UserRecentWorkspace, Long> {
+  List<UserRecentWorkspace> findByUserIdOrderByLastAccessDateDesc(long userId);
+
+  Optional<UserRecentWorkspace> findFirstByWorkspaceIdAndUserId(long workspaceId, long userId);
+}
```

Comment:
For some reason I thought we would be hitting this with a cron job. I don't have any timing data on whether it's worth it, but it would make things a bit easier to see in database logs, firecloud service logs, etc, if it were 1:1. I just have a general aversion to being chatty with the DB.

Reply:
What do you mean by 'hitting this with a cron job' and what were you imagining would be 1:1?